### PR TITLE
ci: migrate CI process to Github actions

### PR DESCRIPTION
**Pull request by mrmodise**
Proposal to migrate to Github actions for the CI/CD process. Configuration is subset of loopback-next

@bajtos @achrinza not sure whether we need test-benchmark for this repo? also, the repetitive build steps on linting. I think it will increase build time unnecessarily

### CODE REVIEW

File: .github/workflows/continous-integration.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      - name: Verify code linting
+        run: npm run lint
+
+  commit-lint:
+    name: Commit Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Un-shallow git tree
+        run: git fetch --prune --unshallow
```
Comment (achrinza) [inline .github/workflows/continous-integration.yaml:75]: We probably should also update the yaml in loopback-next too.

```suggestion
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0
```

**Review by achrinza — CHANGES_REQUESTED** [review]
Great first progress on porting over to GitHub actions! Here's my initial review 👇

**Comment by achrinza** [comment]
> not sure whether we need test-benchmark for this repo?

If there isn't an existing test-benchmark in the Travis CI, then I don't think we need to add one.

> also, the repetitive build steps on linting.

Hmm, you're right; I'm not too sure why we're building before linting. If we can remove it, then that'll be a plus

**Comment by mrmodise** [comment]
PTAL @achrinza

**Comment by mrmodise** [comment]
@achrinza cant start Mongodb on Window nor Macos. See https://github.com/mrmodise/loopback4-example-shopping/pull/3

**Comment by achrinza** [comment]
This seems to be a [known limitation](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idservices) :(

> Note: If your workflows use Docker container actions or service containers, then you must use a Linux runner

We could consider segregating the database integration tests so that we only run it on the ubuntu runners.

**Comment by mrmodise** [comment]
> This seems to be a [known limitation](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idservices) :(
> 
> > Note: If your workflows use Docker container actions or service containers, then you must use a Linux runner
> 
> We could consider segregating the database integration tests so that we only run it on the ubuntu runners.

Ya let me remove the other OS and then create another PR for this

**Comment by mrmodise** [comment]
Found a work-around for Windows 😄

**Comment by mrmodise** [comment]
All green including Windows and MacOs although targeting node 15 only on these:

Screenshot at Jan 09 16-50-18


https://github.com/mrmodise/loopback4-example-shopping/pull/3#issuecomment-757272178

**Comment by achrinza** [comment]
Add my comment here:

https://github.com/mrmodise/loopback4-example-shopping/pull/3#discussion_r554445442

**Review by achrinza — APPROVED** [review]
Overall, LGTM.

Minor nitpick (feel free to ignore):

The arrangement of the OS-specific steps for MongoDB and Redis aren't consistent. So perhaps we could swap the Redis Windows and macOS steps to make it easier to read.

image

**Comment by mrmodise** [comment]
@achrinza i grouped them thanks

**Review by achrinza — APPROVED** [review]
👍

**Comment by raymondfeng** [comment]
@mrmodise Great work. I wonder why this PR does not trigger the github actions.

**Comment by mrmodise** [comment]
@raymondfeng same here but i suspect its some setting on the repo cause i tested on my forked version and it triggered the pipeline: https://github.com/mrmodise/loopback4-example-shopping/pull/3

**Comment by bajtos** [comment]
This is awesome, thank you @mrmodise for the contribution! ❤️

**Pull request merged**